frontend/packages/ceph-storage-plugin/src/components/namespace-store/create-ns.tsx

```diff
@@ -0,0 +1,251 @@
+import * as classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+import * as React from 'react';
+import { ActionGroup, Button, FormGroup, Form, TextInput, Tooltip } from '@patternfly/react-core';
+import {
+  ButtonBar,
+  Dropdown,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import {
+  apiVersionForModel,
+  k8sCreate,
+  referenceForModel,
+  K8sResourceKind,
+} from '@console/internal/module/k8s';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { getName } from '@console/shared';
+import { SecretModel } from '@console/internal/models';
+import { history } from '@console/internal/components/utils/router';
+import { CEPH_STORAGE_NAMESPACE } from '@console/ceph-storage-plugin/src/constants';
+import { NooBaaNamespaceStoreModel } from '../../models';
+import {
+  BC_PROVIDERS,
+  AWS_REGIONS,
+  NOOBAA_TYPE_MAP,
+  PROVIDERS_NOOBAA_MAP,
+  BUCKET_LABEL_NOOBAA_MAP,
+} from '../../constants';
+import {
+  ProviderDataState,
+  S3EndPointType,
+  StoreType,
+  Payload,
+  getProviders,
+  getExternalProviders,
+  StoreAction,
+  secretPayloadCreator,
+} from '../common/noobaa-store/bs-ns-common';
+import '../common/noobaa-store/create-bs-ns.scss';
+
+const PROVIDERS = getProviders(StoreType.NS);
+const externalProviders = getExternalProviders(StoreType.NS);
+
+const initialState: ProviderDataState = {
+  secretName: '',
+  secretKey: '',
+  accessKey: '',
+  region: AWS_REGIONS[0],
+  target: '',
+  endpoint: '',
+};
+
+const providerDataReducer = (state: ProviderDataState, action: StoreAction) => {
+  const { value } = action;
+  switch (action.type) {
+    case 'setSecretName':
+      return Object.assign({}, state, { secretName: value });
+    case 'setSecretKey':
+      return Object.assign({}, state, { secretKey: value });
+    case 'setAccessKey':
+      return Object.assign({}, state, { accessKey: value });
+    case 'setRegion':
+      return Object.assign({}, state, { region: value });
+    case 'setTarget':
+      return Object.assign({}, state, { target: value });
+    case 'setEndpoint':
+      return Object.assign({}, state, { endpoint: value });
+    default:
+      return initialState;
+  }
+};
+
+const CreateNamespaceStoreForm: React.FC<CreateNamespaceStoreFormProps> = withHandlePromise<
```

Comment:
Name the file better , aligned with component name: `namespace-store-form.tsx`

Reply:
changed